ns1/permissions.go

```diff
@@ -235,6 +285,38 @@ func permissionsToResourceData(d *schema.ResourceData, permissions account.Permi
 
 func resourceDataToPermissions(d *schema.ResourceData) account.PermissionsMap {
 	var p account.PermissionsMap
+	if v, ok := d.GetOk("dns_records_allow"); ok {
+		records := []account.Record{}
+		schemaRecord := v.([]interface{})
+		for _, sr := range schemaRecord {
+			mapRecord := sr.(map[string]interface{})
+			record := account.Record{
+				Domain:     mapRecord["domain"].(string),
+				Subdomains: mapRecord["include_subdomains"].(bool),
+				Zone:       mapRecord["zone"].(string),
+				RecordType: mapRecord["type"].(string)}
+			records = append(records, record)
+		}
```

Comment:
You repeat this entire block of logic for records deny below. IMO this is non-trivial because of the mapping of fields and should be made DRY by factoring it out into a separate function.

Reply:
thanks, Done!